exercises/pascals-triangle/example.rs

```diff
@@ -0,0 +1,25 @@
+pub struct PascalsTriangle {
+    row_count: u32,
+}
+
+impl PascalsTriangle {
+    pub fn new(row_count: u32) -> Self {
+        PascalsTriangle { row_count: row_count }
+    }
+
+    pub fn rows(&self) -> Vec<Vec<u32>> {
+        (0..self.row_count).map(|row| PascalsTriangle::row(row)).collect()
+    }
+
+    pub fn row(number: u32) -> Vec<u32> {
+        let mut r = vec![1];
+
+        for p in 1..(number + 1) {
+            if let Some(last) = r.pop() {
+                let next = (last * (number + 1 - p)) / p;
+                r.extend(&[last, next])
```

Comment:
In my testing, `last` returns a borrow, so I can't mutate the vector inside the block.

Example: https://play.rust-lang.org/?gist=4420b3f3b8fa09fbeb78b8eb2b3ca99c&version=stable&backtrace=0